✨Feat: 결제기능, 리뷰기능, 게시글 조회 시 댓글조회 기능 추가

lazzzykim opened the pull request:
## 의도

구매자가 판매자의 게시글에 구매요청을 하고, 결제를 할 수 있습니다.
또, 판매자에게 리뷰를 작성해 평점을 남길 수 있습니다.

- 내용

## 주요 변경점
- 내 계좌에 금액을 충전하고 조회하는 기능 추가
- 판매자의 글에 구매요청 후 결제를 진행하는 로직 작성
- 리뷰를 작성해 판매자를 별점 형태로 평가하는 로직 작성
- 게시글 조회시 댓글이 List로 조회되는 기능 추가


## ETC
리뷰어들에게 어떤 점을 검토해야하는지 알려주세요
- 비효율적인 코드가 많다고 생각됩니다. 리팩터링 할 부분을 집중해 리뷰해주시면 감사하겠습니다.
- 브랜치마다 기능구현시 PR을 작성했어야됬는데, 기능별로 넘어다니며 구현을 하다보니 PR내용이 많습니다. 양해 부탁드립니다.ㅠㅠ

lazzzykim committed:
✨Feat: 게시물 CRUD 구현

PostController: CRUD 요청 메서드 작성
PostResponse: post response 작성
CreatePostRequest: 게시물 create request 코드 작성
UpdatePostRequest: 게시물 update request 코드 작성
Post: post 관련 Entity 작성
PostRepository: PostRepository 생성
PostServiceImpl: 게시물 CRUD 비즈니스 로직 작성

HwangSeungHyeon committed:
✨Feat: 댓글 CRUD 구현

댓글 작성

Resolves: #11

HwangSeungHyeon committed:
✨Feat: 댓글 CRUD 구현

댓글 수정 기능 구현

Resolves: #11

HwangSeungHyeon committed:
✨Feat: 댓글 CRUD 구현

댓글 삭제 기능 구현

Resolves: #11

HwangSeungHyeon committed:
✨Feat: 댓글 CRUD 구현

댓글 삭제 기능

Resolves: #11

lazzzykim committed:
save

lazzzykim committed:
save

lazzzykim committed:
Merge remote-tracking branch 'origin/feature/member' into feature/review

# Conflicts:
#	src/main/kotlin/com/teamsparta/buysell/infra/redis/RedisConfig.kt

lazzzykim committed:
Merge remote-tracking branch 'origin/feature/comment' into feature/review

HwangSeungHyeon committed:
🔨Refactor: request validation 수정

@Size에 field:를 추가

HwangSeungHyeon committed:
✨Feat: 게시글 인가 기능 구현

게시글 작성은 가입된 멤버만 가능하도록 수정
게시글 수정 및 삭제는 작성자만 가능하도록 수정

Resolves: #1

HwangSeungHyeon committed:
✨Feat: 게시글 Pagination

게시글을 조회할 때 QueryDSL을 사용한 pagination 을 구현

Resolves: #1

lazzzykim committed:
Merge remote-tracking branch 'origin/feature/post' into feature/review

lazzzykim committed:
✨Feat: Review Create 구현

CreateReviewRequest: Reqeust 작성
Review: Review Entity 작성
ReviewController: create 요청 메서드 작성
ReviewRepository: Review Repository 생성
ReviewResponse: Response message로 반환하게 작성
ReviewServiceImpl: 리뷰 작성 비즈니스 로직 작성

resolves: #6

lazzzykim committed:
✨Feat: Review Create 구현

CreateReviewRequest: Reqeust 작성
Review: Review Entity 작성
ReviewController: create 요청 메서드 작성
ReviewRepository: Review Repository 생성
ReviewResponse: Response message로 반환하게 작성
ReviewServiceImpl: 리뷰 작성 비즈니스 로직 작성

resolves: #6

HwangSeungHyeon committed:
✨Feat: 게시글 Pagination

게시글 softdelete 구현

Resolves: #1

HwangSeungHyeon committed:
🔨Refactor: soft delete 코드 수정

SoftDeleteEntity를 상속받도록 변경

Resolves: #11

lazzzykim committed:
Merge remote-tracking branch 'origin/feature/comment' into feature/review

lazzzykim committed:
✨Feat: Review Update 구현

ReviewController: Update 요청 메서드 작성
Review: post fetchType.EAGER -> LAZY 변경
ReviewServiceImpl: 리뷰 수정 비즈니스 로직 작성
UpdateReviewRequest: UpdateRequest 작성

resolves: #6

lazzzykim committed:
✨Feat: Review Update 구현

ReviewController: Update 요청 메서드 작성
Review: post fetchType.EAGER -> LAZY 변경
ReviewServiceImpl: 리뷰 수정 비즈니스 로직 작성
UpdateReviewRequest: UpdateRequest 작성

resolves: #6

lazzzykim committed:
Merge remote-tracking branch 'origin/feature/likes' into feature/review

# Conflicts:
#	src/main/kotlin/com/teamsparta/buysell/domain/comment/controller/CommentController.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/comment/dto/response/CommentResponse.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/comment/service/CommentService.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/comment/service/CommentServiceImpl.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/post/controller/PostController.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/post/service/PostService.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/post/service/PostServiceImpl.kt
#	src/main/kotlin/com/teamsparta/buysell/infra/redis/RedisConfig.kt
#	src/main/kotlin/com/teamsparta/buysell/infra/security/jwt/CustomAuthenticationEntryPoint.kt

lazzzykim committed:
✨Feat: Review Delete 구현

Review: Review Entity -> SoftDeleteEntity 상속하게 변경
ReviewController -> Review Delete 요청 메서드 작성
ReviewResponse -> 전체 주석처리 (MessageResponse로 반환)
ReviewServiceImpl -> Delete 로직 작성

resolves: #6

lazzzykim committed:
Merge remote-tracking branch 'origin/feature/review' into feature/review

# Conflicts:
#	src/main/kotlin/com/teamsparta/buysell/domain/review/controller/ReviewController.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/review/dto/response/ReviewResponse.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/review/model/Review.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/review/service/ReviewService.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/review/service/ReviewServiceImpl.kt

lazzzykim committed:
Merge remote-tracking branch 'origin/dev' into feature/review

lazzzykim committed:
Merge remote-tracking branch 'origin/dev' into feature/review

lazzzykim committed:
Refactor: Post - comment 양방향 관계 설정

Comment: Post Entity와 양방향 설정
Post: Comment Entity와 양방향 설정
PostResponse: comment List형태로 response하는 코드추가

resolves: #1

lazzzykim committed:
🔨Refactor: Post - comment 양방향 관계 설정

CommentResponse: comment toResponse 메서드 추가

resolves: #1

lazzzykim committed:
✨Feat: 리뷰 평점기능 추가

CreateReviewRequest: 리뷰관련 validation 추가
Review: rating: Float -> Int로 변경
ReviewController: createReview 메서드에 @Valid 어노테이션 추가
ReviewRepository: AVG 쿼리 작성

resolves: #20

lazzzykim committed:
✨Feat: 계좌 충전,조회 기능 추가

resolves: #8

lazzzykim committed:
✨Feat: 계좌 충전,조회 기능 추가

resolves: #8

lazzzykim committed:
Merge remote-tracking branch 'origin/dev' into feature/account

# Conflicts:
#	src/main/kotlin/com/teamsparta/buysell/domain/member/model/Member.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/member/repository/MemberRepository.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/member/service/MemberServiceImpl.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/post/service/PostServiceImpl.kt
#	src/main/kotlin/com/teamsparta/buysell/infra/security/SecurityConfig.kt

lazzzykim committed:
Merge remote-tracking branch 'origin/dev' into feature/order

lazzzykim committed:
🔨Refactor: Account 주석 제거

resolves: #8

lazzzykim committed:
🔨Refactor: 게시글에서 댓글조회, 게시글 가격수정부분 리팩토링

resolves: #11

lazzzykim committed:
✨Feat: 결제기능 추가

CreateOrderRequest: 주문생성 관련 Request 작성
GlobalExceptionHandler: 잔액이 부족할때 처리하는 IllegalArgumentException 추가
Member: Order 클래스와 연관관계 설정
Order: Member와 N:1관계 설정
OrderController: 주문생성 관련 요청메서드 작성
OrderServiceImpl: 주문생성 비즈니스 로직 작성

resolves: #8

resolves: #11

lazzzykim committed:
🚜Fix: Member Entity - Order Entity 연관관계 수정

1:1관계에서 1:N으로 수정

resolves: #8

lazzzykim committed:
Merge branch 'feature/review' into feature/order

# Conflicts:
#	src/main/kotlin/com/teamsparta/buysell/domain/comment/dto/response/CommentResponse.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/comment/model/Comment.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/post/dto/response/PostResponse.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/post/model/Post.kt

lazzzykim committed:
🔨Refactor: createName 컬럼 삭제

resolves: #8

lazzzykim committed:
✨Feat: 결제기능 추가

CreateOrderRequest: 결제에 필요한 request 작성
Member: 연관관계 1:N으로 연결
Order: 연관관계 N:1로 연결
OrderREpository: 결제관련 repository 작성
OrderServiceImpl: 결제생성 로직 작성

resolves: #8

HwangSeungHyeon commented on src/main/kotlin/com/teamsparta/buysell/domain/member/model/Account.kt:
```diff
@@ -0,0 +1,32 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    var id: Int? = null
+
+    fun modifyBalance(money: Int){
```
계좌 충전에 관한 메서드 이름으로 변경하는 건 어떨까요?

lazzzykim commented on src/main/kotlin/com/teamsparta/buysell/domain/member/model/Member.kt:
```diff
@@ -26,6 +26,13 @@ class Member(
     @Enumerated(EnumType.STRING)
     val role : Role?,
 
+    @OneToOne(cascade = [CascadeType.ALL])
```
같은 판매글에 여러번 결제요청을 했을때 중복결제가 안되어야하게 수정해야합니다.

lazzzykim replied:
다른 회원이 결제를 했을 때, 접근할 수 없게 해야합니다.

lazzzykim commented on src/main/kotlin/com/teamsparta/buysell/domain/post/model/Post.kt:
```diff
@@ -54,7 +64,9 @@ fun Post.toResponse(): PostResponse {
         createdName = member.nickname,
         price = price,
-        isSoldout = isSoldOut
-//        category = category
```
주석 제거

lazzzykim commented on src/main/kotlin/com/teamsparta/buysell/domain/post/service/PostServiceImpl.kt:
```diff
@@ -56,7 +56,7 @@ class PostServiceImpl(
 
         post.title = request.title
         post.content = request.content
-//        post.category = request.category
```
주석처리 해제하기

lazzzykim commented on src/main/kotlin/com/teamsparta/buysell/domain/review/dto/request/CreateReviewRequest.kt:
```diff
@@ -0,0 +1,17 @@
+@Schema(description = "리뷰를 작성할 때 입력한 정보를 전달하는 객체")
+data class CreateReviewRequest(
+
+    @field:NotBlank(message = "내용을 입력해주세요.")
```
글자 수 제한 적용하기

lazzzykim replied:
글자수 제한 적용했습니다.

lazzzykim commented on src/main/kotlin/com/teamsparta/buysell/domain/review/dto/response/ReviewResponse.kt:
삭제

lazzzykim committed:
🔨Refactor: Account Entity에 modifyBalance 메서드명 변경

Account: 계좌충전 관련 함수명을 modifyBalance -> depositToAccount로 변경했습니다.
AccountServiceImpl: 함수명 변경 modifyBalance -> depositToAccount

resolves: #8

codekmj1 reviewed (approved):
확인 했습니다

HwangSeungHyeon committed:
Merge branch 'dev' into feature/order

hyunzoo123123 reviewed (approved):
b'ㅂ'b
